src/shared/bus-wait-for-jobs.c

```diff
@@ -279,14 +279,17 @@ static int check_wait_response(BusWaitForJobs *d, bool quiet, const char* const*
                 return -EOPNOTSUPP;
         else if (streq(d->result, "once"))
                 return -ESTALE;
-        else if (STR_IN_SET(d->result, "done", "skipped"))
+        else if (STR_IN_SET(d->result, "done", "skipped")) {
```

Comment:
If you touch this, could you please move this branch above? The current code mixes success and failure in a big chunk of `else if`s

I.e. something like the following:

```c
if (FLAGS_SET(flags, ..._LOG_ERROR)) {
        ...
}

if (STR_IN_SET(result, "done", "skipped")) {
        if (FLAGS_SET(flags, LOG_SUCCESS))
                ...

        return 0;
}

if (STR_IN_SET(result, "canceled", "collected"))
        ...
...
```

Reply:
Since this is the first contribution of @teknoraver, let's leave the refactoring for another PR